python/mlx/nn/layers/normalization.py

```diff
@@ -178,3 +180,136 @@ def __call__(self, x):
         )
         x = group_norm(x)
         return (self.weight * x + self.bias) if "weight" in self else x
+
+
+class BatchNorm(Module):
+    r"""Applies Batch Normalization over a 2D or 3D input.
+
+    Computes
+
+    .. math::
+
+        y = \frac{x - E[x]}{\sqrt{Var[x]} + \epsilon} \gamma + \beta,
+
+    where :math:`\gamma` and :math:`\beta` are learned per feature dimension
+    parameters initialized at 1 and 0 respectively.
+
+    [1]: https://arxiv.org/abs/1502.03167
+
+    Args:
+        num_features (int): The feature dimension of the input to normalize over.
+        eps (float, optional): A small additive constant for numerical stability. Default is 1e-5.
+        momentum (float, optional): The momentum for updating the running mean and variance. Default is 0.1.
+        affine (bool, optional): If True, learn an affine transform to apply after the normalization. Default is True.
+
+    Examples:
+        >>> import mlx.core as mx
+        >>> import mlx.nn as nn
+        >>> mx.random.seed(42)
+        >>> input = mx.random.normal((5, 4), dtype=mx.float32)
+        >>> # Batch norm
+        >>> bn = nn.BatchNorm(num_features=4, affine=True)
+        >>> output = bn(x)
+    """
+
+    def __init__(
+        self,
+        num_features: int,
+        eps: float = 1e-5,
+        momentum: float = 0.1,
+        affine: bool = True,
+        track_running_stats: bool = True,
+    ):
+        super().__init__()
+
+        self.num_features = num_features
+        self.eps = eps
+        self.momentum = momentum
+        self.affine = affine
+        self.track_running_stats = track_running_stats
+        self._dims_expanded = False
+
+        if self.affine:
+            self.weight = mx.ones((num_features,))
+            self.bias = mx.zeros((num_features,))
+
+        if self.track_running_stats:
+            self.running_mean = mx.zeros((num_features,))
+            self.running_var = mx.ones((num_features,))
```

Comment:
```suggestion
            self._running_mean = mx.zeros((num_features,))
            self._running_var = mx.ones((num_features,))
```
These parameters don't get updated by backprop but are added to the parameters list, add _ to stop gradient computation to not pass [valid_parameter_filter](https://github.com/ml-explore/mlx/blob/cd3616a463a594a1679bd9dc823071bed1469c51/python/mlx/nn/layers/base.py#L112)

otherwise I get `ValueError: [eval] Illegal to eval an array during function transform without graph retention`  during `mx.eval(model.parameters(), optimizer.state)`